hardware_interface/source/hyperion.c

```diff
@@ -67,55 +67,62 @@ void hyperion_config_1_value(config_1_panel_t panel, config_1_channel_type_t cha
         break;
     }
 
-
     switch (channel)
     {
     /* Channel 1 to 3 handle temp sensor */
     case CONFIG_1_CHANNEL_TEMP_1:
         adc_init(slave_addr, 1 << 7);
-        adc_get_raw(slave_addr, &data, &ch);
+        adc_get_raw(slave_addr, &data, &ch); // invalid value
+        adc_get_raw(slave_addr, &data, &ch); // correct value
```

Comment:
Why do you do this twice?

Reply:
The first read is invalid according to this. 
https://github.com/AlbertaSat/ex2_hyperion_solar_panel_software/blob/cd8c4c6db6fa4c005760c3489d16796ce79db181/equipment_handler/source/adc_handler.c#L158